src/server/rdb_save.cc

```diff
@@ -1022,7 +1024,7 @@ error_code RdbSaver::Impl::ConsumeChannel(const Cancellation* cll) {
   // we can not exit on io-error since we spawn fibers that push data.
   // TODO: we may signal them to stop processing and exit asap in case of the error.
 
-  while (record = records_popper.Pop()) {
+  while ((record = records_popper.Pop())) {
```

Comment:
You could also do:

```cpp
while (record = records_popper.Pop(); record.has_value()) {
```

or

```cpp
while (record = records_popper.Pop(); record) {
```

Your call, but I think the double parenthesis can be confusing

Reply:
I just fixed warnings by compiler and did what he told me to :)